Source/WebKit/UIProcess/API/Swift/WebPage.swift

```diff
@@ -24,13 +24,80 @@
 #if ENABLE_SWIFTUI
 
 import Foundation
+import Observation
+import SwiftUI // FIXME: Do not import SwiftUI in WebKit proper.
 
 @_spi(Internal)
-@MainActor public class WebPage_v0 {
+@MainActor
+@Observable
+public class WebPage_v0 {
     public let navigations: Navigations
 
+    public var url: URL? {
+        accessBackingProperty(\.url, backedBy: \.url)
+    }
+
+    public var title: String {
+        // The title property is annotated as optional in WKWebView, but is never actually `nil`.
+        _$observationRegistrar.access(self, keyPath: \.title)
+        return backingWebView.title!
+    }
+
+    public var estimatedProgress: Double {
+        accessBackingProperty(\.estimatedProgress, backedBy: \.estimatedProgress)
+    }
+
+    public var isLoading: Bool {
+        accessBackingProperty(\.isLoading, backedBy: \.isLoading)
+    }
+
+    public var serverTrust: SecTrust? {
+        accessBackingProperty(\.serverTrust, backedBy: \.serverTrust)
+    }
+
+    public var hasOnlySecureContent: Bool {
+        accessBackingProperty(\.hasOnlySecureContent, backedBy: \.hasOnlySecureContent)
+    }
+
+    public var themeColor: Color? {
+        // The color property is a UIColor/NSColor in WKWebView.
+        _$observationRegistrar.access(self, keyPath: \.themeColor)
+        return backingWebView.themeColor.map(Color.init(_:))
+    }
+
+    public var mediaType: String? {
```

Comment:
IMO, single line `get` and `set` lines (and even implicit `get`s like in the above) can be written as a single line too.
```swift
public var mediaType: String? {
    get { backingWebView.mediaType }
    set { backingWebView.mediaType = newValue }
}
```

Why do these properties not need to also invoke the registrar for change tracking?

Reply:
because they are not observable / kvo compliant

Reply:
Hmm, we might need to consider this more. Having a few one off non-observable properties is going to be confusing to adopters. I don’t think there’s anything to stop you from doing something like:
```swift
@Bindable var webPage = self.webPage
TextField(“Media Type:”, text: $webPage.mediaType)
```
(Ignoring that `Binding<String?>` can’t be passed to a text field, but the point stands).